### ISSUE TITLE
Bug with Beating Embryo

### ISSUE THREAD

**Issue by Mysticott** [ThinkInvis/RoR2-ClassicItems#99]
### Describe the Bug
When using the equipment it doesn't seem to be 30% chance nor does it double the effects rather multiplies the effects by I'd say 30.

### To Reproduce
1. Grab a Beating Embryo
2. Grab any equipment.
3. Activate it once.

### Expected Behavior
I expected either 2 chest to fall out of the sky rather then a whole set of chests.
![1](https://user-images.githubusercontent.com/40555923/159997590-6e9ea9dd-0baa-41e8-9927-ba539feead09.png)
![2](https://user-images.githubusercontent.com/40555923/159997616-ca83da6a-236f-4d4b-8867-9214b061c58e.png)

**Comment by ThinkInvis** [comment]
Can't reproduce. Console log, please.

**Comment by aaomidi** [comment]
Interesting I got a weird bug but I hadn't even picked up that item:

```
[Error  : Unity Log] NullReferenceException
Stack trace:
UnityEngine.Component.GetComponent[T] () (at <469e86e97fa2467ebffe49bbd6086325>:IL_0021)
ThinkInvisible.ClassicItems.Embryo.CheckLastEmbryoProc (RoR2.EquipmentSlot slot) (at <79df9eec8338472f87fa54f4899b3366>:IL_001D)
ThinkInvisible.ClassicItems.EmbryoHooks.LifestealOnHit+<>c__DisplayClass9_0.<GlobalEventManager_OnHitEnemy>b__0 (RoR2.EquipmentSlot slot) (at <79df9eec8338472f87fa54f4899b3366>:IL_0000)
MonoMod.Cil.RuntimeILReferenceBag+FastDelegateInvokers.Invoke[T1] (T1 arg1, MonoMod.Cil.RuntimeILReferenceBag+FastDelegateInvokers+Action`1[T1] del) (at <6733e342b5b549bba815373898724469>:IL_0000)
RoR2.GlobalEventManager.OnHitEnemy (RoR2.DamageInfo damageInfo, UnityEngine.GameObject victim) (at <0f44b36eaa194934bb127b3fdd08e444>:IL_000B)
DMD<>?-1999458048.Trampoline<RoR2.GlobalEventManager::OnHitEnemy>?-1554265088 (RoR2.GlobalEventManager , RoR2.DamageInfo , UnityEngine.GameObject ) (at <ac0a0f1df9624e5fa46bf0aaa3b9970c>:IL_0020)
ShareSuite.MoneySharingHooks.BrittleCrownOnHitHook (On.RoR2.GlobalEventManager+orig_OnHitEnemy orig, RoR2.GlobalEventManager self, RoR2.DamageInfo info, UnityEngine.GameObject victim) (at <00ea748bcf0e4a9db6440754dfb0ddd4>:IL_0077)
DMD<>?-1999458048.Hook<RoR2.GlobalEventManager::OnHitEnemy>?-403283712 (RoR2.GlobalEventManager , RoR2.DamageInfo , UnityEngine.GameObject ) (at <2defcad959bd457f84f0308c4e88254d>:IL_000A)
DMD<>?-1999458048.Trampoline<RoR2.GlobalEventManager::OnHitEnemy>?720709632 (RoR2.GlobalEventManager , RoR2.DamageInfo , UnityEngine.GameObject ) (at <1edbaef0dbd04510ae9e13efb80c9f74>:IL_0020)
MysticsRisky2Utils.GenericGameEvents+<>c.<Init>b__40_1 (On.RoR2.GlobalEventManager+orig_OnHitEnemy orig, RoR2.GlobalEventManager self, RoR2.DamageInfo damageInfo, UnityEngine.GameObject victim) (at <268c0c739f2845c5aa626e8f26acb321>:IL_0001)
DMD<>?-1999458048.Hook<RoR2.GlobalEventManager::OnHitEnemy>?-1128203648 (RoR2.GlobalEventManager , RoR2.DamageInfo , UnityEngine.GameObject ) (at <3b02f51005304d959df01d6fd5e3bfb2>:IL_0014)
DMD<>?-1999458048.Trampoline<RoR2.GlobalEventManager::OnHitEnemy>?307383296 (RoR2.GlobalEventManager , RoR2.DamageInfo , UnityEngine.GameObject ) (at <b8c232616ed44fa68b46fc2765455448>:IL_0020)
ThinkInvisible.ClassicItems.Pillage.On_GEMOnHitEnemy (On.RoR2.GlobalEventManager+orig_OnHitEnemy orig, RoR2.GlobalEventManager self, RoR2.DamageInfo damageInfo, UnityEngine.GameObject victim) (at <79df9eec8338472f87fa54f4899b3366>:IL_0000)
DMD<>?-1999458048.Hook<RoR2.GlobalEventManager::OnHitEnemy>?-1628445184 (RoR2.GlobalEventManager , RoR2.DamageInfo , UnityEngine.GameObject ) (at <e1757458025d4727a63cfb6491f186e4>:IL_0014)
DMD<>?-1999458048.Trampoline<RoR2.GlobalEventManager::OnHitEnemy>?1531797504 (RoR2.GlobalEventManager , RoR2.DamageInfo , UnityEngine.GameObject ) (at <74e8aeef372b493eaac393262ebd4f90>:IL_0020)
ThinkInvisible.ClassicItems.TeleSight.On_GEMOnHitEnemy (On.RoR2.GlobalEventManager+orig_OnHitEnemy orig, RoR2.GlobalEventManager self, RoR2.DamageInfo damageInfo, UnityEngine.GameObject victim) (at <79df9eec8338472f87fa54f4899b3366>:IL_0000)
DMD<>?-1999458048.Hook<RoR2.GlobalEventManager::OnHitEnemy>?1620599040 (RoR2.GlobalEventManager , RoR2.DamageInfo , UnityEngine.GameObject ) (at <c5afb5cb714441929fc0b7262af244e9>:IL_0014)
RoR2.Projectile.ProjectileSingleTargetImpact.OnProjectileImpact (RoR2.Projectile.ProjectileImpactInfo impactInfo) (at <0f44b36eaa194934bb127b3fdd08e444>:IL_0189)
RoR2.Projectile.ProjectileController.OnCollisionEnter (UnityEngine.Collision collision) (at <0f44b36eaa194934bb127b3fdd08e444>:IL_0059)
```

The enemy may have been the one who had this item. Anyway yeah, it spams the console and kills the game eventually.

**Comment by aaomidi** [comment]
https://github.com/ThinkInvis/RoR2-ClassicItems/blob/master/Items/T3/Embryo.cs#L154-L159 You might want to check if the component exists first - or alternatively a try/catch around the getComponent call. I can spin up a PR, but I'm not super familiar with C#.

**Comment by ThinkInvis** [comment]
L157 and L155 *should* be null checks already: Unity overrides the ! operator on components/GameObjects to do so. I'll make it an explicit check to see if that fixes it.

**Comment by ThinkInvis** [comment]
Never mind, spotted why that's happening (copy-pasted some IL into a method with a mismatched signature): https://github.com/ThinkInvis/RoR2-ClassicItems/blob/64d4ce60f8f2779b187e011acf132a15d9839533/Items/T3/VanillaEmbryoHooks/LifestealOnHit.cs#L26-L33

**Comment by ThinkInvis** [comment]
Appears to be fixed in 6.1.1 (could not cause issue after triggering LifestealOnHit equipment, with or without any stacks of Embryo).

TIL `ComponentType1.GetComponent<ComponentType2>` throws a NRE if you feed it an object that isn't actually of the first component type, and not something more useful... like, at least give me an ArgumentException, that'd be a little more accurate :(

**Comment by aaomidi** [comment]
Yeah that was certainly an odd error. Cheers on the fix!
